alma/views/js/alma-inpage.js

```diff
@@ -22,215 +22,166 @@
-                                }
-                                $('.alma-loader--wrapper').remove();
-                            }
-                        }
-                    );
-                })
-                .fail(function () {
-                    let pathnameUrl = window.location.pathname;
-                    let urlError = "/order";
-
-                    if (pathnameUrl === '/index.php') {
-                        urlError = "/index.php?controller=order";
-                    }
-
-                    window.location.href = urlError;
-                });
-        }
-    }
-
-    const isAlmaPayment = function (url) {
-        return url.indexOf("module/alma/payment") !== -1 || url.indexOf("module=alma") !== -1;
-    };
-
-    const getURLParameter = function(sUrl, sParam)
-    {
-        var sURLVariables = sUrl.split('?');
-        for (var i = 0; i < sURLVariables.length; i++)
-        {
-            var sParameterName = sURLVariables[i].split('=');
-            if (sParameterName[0] == sParam)
-            {
-                return decodeURIComponent(sParameterName[1]);
+            if( inPage !== undefined) {
+                inPage.unmount();
             }
-        }
-    };
-
-    const createIframe = function (paymentOptionId, selectorSetting, showPayButton, url = '') {
-        let merchantId = selectorSetting.data('merchantid');
-        let installment = selectorSetting.data('installment');
-        let purchaseAmount = selectorSetting.data('purchaseamount');
-        let locale = selectorSetting.data('locale');
-
-        selectorSetting.attr('id', 'alma-inpage-' + paymentOptionId);
 
-        let selectorIframeInPage = '#alma-inpage-' + paymentOptionId;
+            createAlmaIframe(settingInpage, true, url);
+        });
+    });
+}
 
-        if (showPayButton) {
-            // No refactor inPage is use in callback function 1.6
-            inPage = Alma.InPage.initialize(
-                {
-                    merchantId: merchantId,
-                    amountInCents: purchaseAmount,
-                    installmentsCount: installment,
-                    locale: locale,
-                    environment: selectorSetting.data("apimode"),
-                    selector: selectorIframeInPage,
-                    onIntegratedPayButtonClicked : () => {
-                        ajaxPayment(url, inPage);
-                    }
-                }
-            );
+function createAlmaIframe(form, showPayButton = false, url = '') {
+    let merchantId = form.dataset.merchantid;
+    let installment = form.dataset.installment;
+    let purchaseAmount = form.dataset.purchaseamount;
+    let locale = form.dataset.locale;
 
-            return inPage;
-        }
+    let selectorIframeInPage = form.querySelector('.alma-inpage-iframe');
 
-        return Alma.InPage.initialize(
+    if (showPayButton) {
+        // We couldn't refactor. inPage is used in callback function 1.6
```

Comment:
What exactly were you not able to do? ("We couldn't refactor." is too vague)

Reply:
It's because we use twice Alma.Inpage.initialize in this function. I prefer remove de comment :)